### PR TITLE
[GH-13099] Migrate lastPostTimeCache cache from store/sqlstore/post_store.go to the new store/localcachelayer

### PR DESCRIPTION
**Pull request by larkox**
#### Summary

Migrate lastPostTimeCache cache from store/sqlstore/post_store.go to the new store/localcachelayer
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/13099

### CODE REVIEW

**Comment by jespino** [comment]
@larkox please, fix the merge conflicts here.

File: store/sqlstore/post_store.go [hunk 1]
```diff
@@ -560,8 +509,6 @@ func (s *SqlPostStore) GetPostsSince(options model.GetPostsSinceOptions, allowFr
 		}
```
Comment (jespino) [inline store/sqlstore/post_store.go]: all the references to latestUpdate before this line in this function should be removed, are only cache related so you don't need it here now.

File: store/localcachelayer/post_layer.go [hunk 2]
```diff
@@ -23,26 +35,75 @@ func (s *LocalCachePostStore) handleClusterInvalidateLastPosts(msg *model.Cluste
...
+	latestUpdate := options.Time
+	if err == nil {
+		for _, p := range list.ToSlice() {
+			if latestUpdate < p.UpdateAt {
+				latestUpdate = p.UpdateAt
+			}
+		}
+	}
+
+	s.rootStore.doStandardAddToCache(s.rootStore.lastPostTimeCache, options.ChannelId, latestUpdate)
```
Comment (jespino) [inline store/localcachelayer/post_layer.go]: It should store the latestUpdate value in the cache only if there is not error.

File: store/localcachelayer/post_layer_test.go [hunk 3]
```diff
@@ -1,9 +1,10 @@
-// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
```
Comment (jespino) [inline store/localcachelayer/post_layer_test.go]: You don't need this change.

File: store/localcachelayer/post_layer.go [hunk 4]
```diff
@@ -23,26 +35,75 @@ func (s *LocalCachePostStore) handleClusterInvalidateLastPosts(msg *model.Cluste
...
+		if lastTime := s.rootStore.doStandardReadCache(s.rootStore.lastPostTimeCache, channelId); lastTime != nil {
+			return fmt.Sprintf("%v.%v", model.CurrentVersion, lastTime.(int64))
+		}
+	}
+
+	result := s.PostStore.GetEtag(channelId, allowFromCache)
+
+	splittedResult := strings.Split(result, ".")
+
+	lastTime, _ := strconv.ParseInt((splittedResult[len(splittedResult)-1]), 10, 64)
```
Comment (jespino) [inline store/localcachelayer/post_layer.go]: This is not the same behavior and I don't like it. Anyway, I thinking we should cache the entire result string, so we don't need to parse and build the string each single time.

Comment (larkox) [reply store/localcachelayer/post_layer.go]: Other changes corrected. For this, in order to cache the string, we would need a new cache. Is this what we want?
If so, is there any possibility that `model.CurrentVersion` changes during the lifetime of the cache? If so, we would need to invalidate this etag on those cases.

Comment (jespino) [reply store/localcachelayer/post_layer.go]: No, It is not going to change until you upgrade the server, and in that cases makes a lot of sense to invalidate the cache anyway.

Comment (jespino) [reply store/localcachelayer/post_layer.go]: I think we should change how we handle the GetEtag from the store entirely, changing the GetEtag calculation outside of the store. That doesn't have anything to do with the store itself, we need to get the data from the store needed to generate the Etag, but not the etag itself. I think we can go forward with this approach and later on change the GetEtag function entirely with something like GetLastPostDate, or something like that.

File: store/localcachelayer/post_layer_test.go [hunk 5]
```diff
@@ -17,6 +18,113 @@ func TestPostStore(t *testing.T) {
 	StoreTestWithSqlSupplier(t, storetest.TestPostStore)
 }
 
+func TestPostStoreLastPostTimeCache(t *testing.T) {
+	var fakeLastTime int64 = 1
+	//fakeOptions :=
```
Comment (mgdelacroix) [inline store/localcachelayer/post_layer_test.go]: This comment can be removed

**Pull request merged**